spark/src/test/resources/tpcds-query-results/v1_4/q39a.sql.out

```diff
@@ -31,7 +31,7 @@ struct<w_warehouse_sk:int,i_item_sk:int,d_moy:int,mean:double,cov:double,w_wareh
 1	12259	1	326.5	1.219693210219279	1	12259	2	292.6666666666667	1.2808898286830026
 1	12641	1	321.25	1.1286221893301993	1	12641	2	279.25	1.129134558577743
 1	13043	1	260.5	1.355894484625015	1	13043	2	295.0	1.056210118409035
-1	13157	1	260.5	1.5242630430075292	1	13157	2	413.5	1.0422561797285326
+1	13157	1	260.5	1.524263043007529	1	13157	2	413.5	1.0422561797285326
```

Comment:
Wondering what is causing the digit difference...

Reply:
I  am not sure what caused the digit difference.
Actually for SortMergeJoin, I got `1.524263043007529`, but for BroadCastJoin, I still got `1.5242630430075292`. Is it OK if I change the the expected result based on join type?

Reply:
also cc @viirya

Reply:
Ideally, it would be good to compare floating point numbers based on an epsilon to make sure they are within some tolerance threshold. I assume we are currently just comparing text file output directly? Do we have a way to generate the output into a structured file type such as CSV or JSON?

Reply:
The difference may be down to order of operations - depending on the order that batches that are being processed from different partitions, for example. I don't think we can expect it to be 100% deterministic in a distributed system.

Reply:
Yes, we are currently just comparing text file output directly. We are using Spark's TPCDSQuerySuite. It doesn't seem to be a way to generate the output into a structured file type.